platform/lib/build/samplesBuilder.js

```diff
@@ -383,13 +386,28 @@ class SamplesBuilder {
       }
     }
 
+    // Sort component samples to always have the specific component
+    // sample at first
+    for (const component of Object.keys(this._componentSamples)) {
+      this._componentSamples[component] = Object.values(this._componentSamples[component]);
+
+      this._componentSamples[component].sort((sample1, sample2) => {
+        return sample1.title.startsWith(component) ? 1 : 0;
+      }).reverse();
+    }
+
     try {
       await writeFileAsync(SITEMAP_DEST, JSON.stringify(this._sitemap), {
-        flag: 'wx+',
+        flag: 'w+',
+      });
+
+      await writeFileAsync(COMPONENT_SAMPLES_DEST, JSON.stringify(this._componentSamples), {
+        flag: 'w+',
       });
-      this._log.success('Wrote sample sitemap.');
+
+      this._log.success('Wrote sample sitemap and component samples file.');
     } catch (_) {
-      this._log.info('Samples sitemap already exists');
+      this._log.error('Writing samples builder meta files failed:', e);
```

Comment:
I know this was the same before, but wouldn't it be better to fail the build here?